src/backend/base/langflow/services/queue/service.py

```diff
@@ -0,0 +1,153 @@
+from __future__ import annotations
+
+import asyncio
+import contextlib
+
+from loguru import logger
+
+from langflow.events.event_manager import EventManager, create_default_event_manager
+from langflow.services.base import Service
+
+
+class QueueService(Service):
+    name = "queue_service"
+
+    def __init__(self) -> None:
+        self._queues: dict[str, tuple[asyncio.Queue, EventManager, asyncio.Task | None]] = {}
+        self._cleanup_task: asyncio.Task | None = None
+        self._closed = False
+
+    async def start(self) -> None:
+        """Start the queue service and its cleanup task."""
+        self._closed = False
+        self._cleanup_task = asyncio.create_task(self._periodic_cleanup())
+
+    async def stop(self) -> None:
+        """Stop the queue service and cleanup all resources."""
+        self._closed = True
+        if self._cleanup_task:
+            self._cleanup_task.cancel()
+            with contextlib.suppress(asyncio.CancelledError):
+                await self._cleanup_task
```

Comment:
```suggestion
            self._cleanup_task.cancel()
            await asyncio.wait([self._cleanup_task])
            if not self._cleanup_task.cancelled():
                exc = self._cleanup_task.exception()
                if exc is not None:
                    raise exc
```
CancelledError should always be propagated. Here `CancelledError` could be raised because the parent was cancelled, not the child task.